ekundayo-ab/hello-books
# #149786086 Admin Borrow Return Notification

### What does this PR do?
At the core of its implementation, It ensures that an Administrative User gets real time notifications when logged whenever a book is borrowed.

If Such Admin User is not logged in, cumulative notifications can still be seen on the dashboard

### Description of Task to be completed?
1. Setup package for real time communications (**socket.io**)
2. Make **socket.io** listen on the main server host and port
3. Setup notification menu and notification list on Administrative dashboard
4. Write logic to notify only Administrative User when any authenticated user borrows a book
5. Do same as in No. 4, but also when such user returns the borrowed book

### How should this be manually tested?
1. Pull in latest changes from remote repository to local repository
2. Fire up terminal run `npm install`
3. execute `npm run start:dev` and then `npm run build:dev` to start up application
4. Visit url supplied by the application usually **http://localhost:3000**
5. login with `ekprogs@gmail.com` and `dayo` for admin access, while for normal user access simply register a new user on the application
6. Open two browsers (preferably **firefox** and **chrome**) login as admin on one and normal user on another
7. Borrow a book as a normal user and then navigate to the admin user browser and see notification in action, also click on notification bell to see notification lists on the admin dashboard

### Any background context you want to provide?
N/A

### What are the relevant pivotal tracker stories?
#149786086 Admin User should get notifications when a book is borrowed or returned

### Screenshots (if appropriate)
<img width="1436" alt="notification_menu" src="https://user-images.githubusercontent.com/20375577/33758005-d3c5b58a-dbfc-11e7-9ac7-4b4d40e1d837.png">
<img width="1440" alt="notification_list" src="https://user-images.githubusercontent.com/20375577/33758018-d9ddcfd4-dbfc-11e7-9f07-b93f95fc9b1c.png">

### Questions:
N/A
